e2echart/e2e-chart-template.html

```diff
@@ -405,43 +634,101 @@ <h5 class="modal-title">Resource</h5>
         }
     }
 
-    function renderChart(regex) {
+    function renderChart() {
         var loc = window.location.href;
 
-        var timelineGroups = []
-        timelineGroups.push({group: "operator-unavailable", data: []})
-        createTimelineData("OperatorUnavailable", timelineGroups[timelineGroups.length - 1].data, eventIntervals, isOperatorAvailable, regex)
+        var isSet = false
+        var positiveSelectionRows = new Map();
+        for (let i = 0; i < numPositiveSelectors; i++){
+            var category = $("#category_"+i).val()
+            var lodash = $("#lodash_"+i).val()
+            var ns = $("#namespace_"+i).val()
+            var regexStr = $("#filterInput_"+i).val()
+            var currIsSet = (regexStr.length != 0 || lodash.length != 0 ||  ns.length != 0 || category.length != 0)
+            positiveSelectionRows.set(i,
+                {
+                    isSet: currIsSet,
+                    lodash: lodash,
+                    regexStr: regexStr,
+                    regex: new RegExp(regexStr),
+                    namespace: ns,
+                    category: category
+                })
+
+            if (currIsSet) {
+                isSet = true
+            }
+        }
 
-        timelineGroups.push({group: "operator-degraded", data: []})
-        createTimelineData("OperatorDegraded", timelineGroups[timelineGroups.length - 1].data, eventIntervals, isOperatorDegraded, regex)
+        var filteredEvents;
+        // if none of the inputs are set, nothing to filter so don't waste time looping through everything
+        if (!isSet) {
+            filteredEvents = eventIntervals.items;
+        }
+        else {
+            // At least one of the inputs had a value, test any inputs that had values
+            // This currently does an OR operation of the input fields
+            filteredEvents = _.filter(eventIntervals.items, function(eventInterval) {
+                // Go ahead and filter out uncategorized events
+                if (eventInterval.categories.uncategorized) {
+                    return false;
+                }
+                var eventMatches = false;
+
+                for (let [key, positiveSelectionRow] of positiveSelectionRows) {
+                    if (positiveSelectionRow.isSet){
+                        matchRegex = positiveSelectionRow.regexStr.length == 0 || positiveSelectionRow.regex.test(eventInterval.locator)
+                        matchNS = positiveSelectionRow.namespace.length == 0 || eventInterval.locatorObj.ns && eventInterval.locatorObj.ns.includes(positiveSelectionRow.namespace)
+                        matchCategory = positiveSelectionRow.category.length == 0 || eventInterval.categories[positiveSelectionRow.category]
+                        matchLodash = true
+                        if (positiveSelectionRow.lodash.length > 0){
+                            lodashMatches = _.filter([eventInterval], _.matches(JSON.parse(positiveSelectionRow.lodash)));
```

Comment:
JSON.parse will easily throw exceptions here as someone as typing in their json string, you need to wrap JSON.parse in a try catch.

```
   var parsedLodash;
    try {
        parsedLodash = JSON.parse(positiveSelectionRow.lodash);
    } catch(e) {
        // whatever you want to do with the error handling to inform user its not valid
        // like change the input's outline to red
    }
```